inst/sympref.m

```diff
@@ -183,140 +183,162 @@
     return
   end
 
-
-  switch lower(cmd)
-    case 'defaults'
-      settings = [];
-      settings.ipc = 'default';
-      settings.whichpython = '';
-      sympref ('display', 'default')
-      sympref ('digits', 'default')
-      sympref ('quiet', 'default')
-
-    case 'version'
-      assert (nargin == 1)
-      varargout{1} = '2.5.0-dev';
-
-    case 'display'
-      if (nargin == 1)
-        varargout{1} = settings.display;
+  if (isstruct(cmd))
+    % Means we need to copy it
+    s = sympref();
+    names = fieldnames (s);
+
+    if (sum(isfield(cmd, names)) == numfields(s))
+      if (typeinfo(cmd.ipc) == typeinfo(s.ipc) && ...
+        typeinfo(cmd.whichpython) == typeinfo(s.whichpython))
+        settings = [];
+        settings.ipc = cmd.ipc;
+        settings.whichpython = cmd.whichpython;
+        sympref ('display', cmd.display)
+        sympref ('digits', cmd.digits)
+        sympref ('quiet', cmd.quiet)
```

Comment:
some of these set the `settings` structure and others call `sympref`.  Why?  Probably needs some comment.

I also expect that `ipc` will have to be called (`sympref ('ipc', cmd.ipc)`)